test/test_cuda.py

```diff
@@ -1314,6 +1314,20 @@ def test_record_stream_on_shifted_view(self):
 
         self.assertNotEqual(try_realloc.data_ptr(), data_ptr)
 
+    def test_external_streams(self):
+        stream_a = torch.cuda.Stream()
+        ext_stream = torch.cuda.streams.ExternalStream(stream_a.cuda_stream)
+        self.assertEqual(stream_a.cuda_stream, ext_stream.cuda_stream)
+
+    @unittest.skipIf(not TEST_MULTIGPU, "detected only one GPU")
+    def test_external_streams_multi_device(self):
+        device = torch.device('cuda:1')
+        stream_a = torch.cuda.Stream(device=device)
```

Comment:
do you want to have tests with real external streams, and not native streams masquerading as external?
```
lib = ctypes.cdll.LoadLibrary(None)
p=ctypes.c_void_p()
out=lib.cudaStreamCreate(ctypes.byref(p))
```

Reply:
this is pretty cool, thanks!!!